app/derivative-services/jp2_derivative_service.rb

```diff
@@ -114,7 +114,30 @@ def use
   end
 
   def filename
-    return Pathname.new(file_object.io.path) if file_object.io.respond_to?(:path) && File.exist?(file_object.io.path)
+    Pathname.new(file_object.disk_path)
+  end
+
+  def clean_filename
+    @filename ||=
+      begin
+        Pathname.new(cleaned_file.path)
+      rescue MiniMagick::Error
+        filename
+      end
+  end
+
+  # Remove compression from given TIFF file, just in case.
+  def cleaned_file
+    @cleaned_file ||=
+      begin
+        t = Tempfile.new
```

Comment:
Do we need to explicitly clean up this tempfile after the derivative processing happens?

Reply:
No, Tempfile means it should get deleted after the process closes.